core/src/main/java/com/graphhopper/matrix/algorithm/MatrixAlgorithm.java

```diff
@@ -0,0 +1,39 @@
+package com.graphhopper.matrix.algorithm;
+
+import com.graphhopper.matrix.DistanceMatrix;
+
+import java.util.List;
+
+/**
+ * Represents a distance matrix algorithm.
+ *
+ * @author Pascal Büttiker
+ */
+public interface MatrixAlgorithm {
+
+    // Known named implementations:
+
+    /**
+     * A very simplistic implementation which invokes a
+     * standard route() algorithm for every node to node route.
+     *
+     * Results in quadratic complexity on-top of the underlying route algorithm.
+     * Not very efficient.
+     */
+    String OneToOneLoop = "one-to-one-Loop";
+
+    /**
+     * This implementation uses for each node a one-to-many Dijkstra
+     */
+    String OneToManyDijkstra = "one-to-many-Dijkstra";
```

Comment:
I didnt use the AlgorithmOptions since the MatrixAlgorithm & Factory was completely separated, but probably we can have them in there any way. Maybe we just prefix matrix algorithms with `Matrix`?
I.e.:  `MatrixOneToOneLoop` ?


As a side note: I wonder if it is a good idea to have the algorithm configurable for the matrix. We could probably choose the best algorithm automatically based on Graph Size and query input?